src/main/java/org/alfresco/util/transaction/TransactionSupportUtil.java

```diff
@@ -130,117 +112,131 @@ public static boolean isActualTransactionActive()
     }
     
     /**
-     * Gets a resource associated with the current transaction, which must be active.
-     * <p>
-     * All necessary synchronization instances will be registered automatically, if required.
-     * 
+     * Gets a resource associated with the current transaction
      *  
      * @param key the thread resource map key
      * @return Returns a thread resource of null if not present
-     * 
-     * @see org.alfresco.repo.transaction.TransactionalResourceHelper         for helper methods to create and bind common collection types
      */
     @SuppressWarnings("unchecked")
-    public static <R extends Object> R getResource(Object key)
+    public static <R> R getResource(Object key)
     {
-        // get the synchronization
-        TransactionSynchronizationImpl txnSynch = getSynchronization();
-        // get the resource
-        Object resource = txnSynch.resources.get(key);
-        // done
+        if (TransactionSynchronizationManager.isSynchronizationActive())
```

Comment:
Please add a comment on why this block of code is necessary.

Reply:
Good point. My first thought was that this check is not necessary, but I soon realized that it is not true.
The comments were added.